main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordDeclCompleter.scala

```diff
@@ -0,0 +1,37 @@
+
+/*
+ * Copyright 2023 Magnus Madsen, Alexander Dybdahl Troelsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.Index
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+object KeywordDeclCompleter extends Completer {
+
+  def getCompletions(context: CompletionContext)(implicit flix: Flix, index: Index, root: TypedAst.Root): Iterable[Completion] = 
+    List(      
+      "def",
+      "eff",
+      "enum",
+      "instance",
+      "mod",
+      "pub",
```

Comment:
This is a modifier. So it raises the question if declarations can be further subdivided.

Reply:
as in not all declarations can be made made public so not all decl keyword can follow `pub`?

Reply:
As in a modifier and a declaration are not the same type of icecream...

Reply:
Yeah. I was just thinking about the how the difference can be leveraged to give better suggestions. I was thinking there was little reason to subdivide further if it would not enable better suggestions, hence why I went directly to the difference between the context in which they appear. But yeah, ofc. it's not the same kind of keyword.

I think I do want subdivide further here :)

Reply:
I feel pretty good about the subdivision at this point, so I'll say this one is resolved